llvm/lib/CodeGen/TargetInstrInfo.cpp

```diff
@@ -578,28 +578,37 @@ static void foldInlineAsmMemOperand(MachineInstr *MI, unsigned OpNo, int FI,
     foldInlineAsmMemOperand(MI, TiedTo, FI, TII);
   }
 
-  // Change the operand from a register to a frame index.
-  MO.ChangeToFrameIndex(FI, MO.getTargetFlags());
-
-  SmallVector<MachineOperand, 4> NewOps;
-  TII.getFrameIndexOperands(NewOps);
+  SmallVector<MachineOperand, 5> NewOps;
+  TII.getFrameIndexOperands(NewOps, FI);
```

Comment:
The old behavoir is to change to frame index. Does it mean we need to call this method in `TargetInstrInfo::getFrameIndexOperands` for other targets?

Reply:
Yes, and the previous behavior did already call `TargetInstrInfo::getFrameIndexOperands`.  Instead of `MachineOperand::ChangeToFrameIndex` + `TargetInstrInfo::getFrameIndexOperands`, this change instead does `TargetInstrInfo::getFrameIndexOperands` (which now creates the MachineOperand we were changing) then `MachineInstr::removeOperand` (to remove the operand we were previously changing).

I have additional child commits locally for supporting other targets.  I need to rebase them onto this change, which shouldn't be a problem; basically `getFrameIndexOperands` just needs to set up 3 MachineOperands for most targets but 5 MachineOperands for x86.  Even then, I've realized that we only need to support x86 and m68k.